### PR TITLE
Add OpenWithCreate(...) as a proof of concept.

### PR DESCRIPTION
**Pull request by mitchdenny**
Related to #4160 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4162)

### CODE REVIEW

File: src/Components/Aspire.Microsoft.Data.SqlClient/AspireSqlServerSqlClientExtensions.cs [hunk 1]
```diff
@@ -19,6 +19,134 @@ public static class AspireSqlServerSqlClientExtensions
...
+    /// This method wraps <see cref="Microsoft.Data.SqlClient.SqlConnection.Open()"/> but adds exception handling to detect
+    /// a <see cref="SqlException"/> with a <see cref="SqlException.Number"/> of 4060 (Database not found). If this exception is
+    /// throw the connection is closed and the connection string updated to connect to the <c>master</c> database.
+    /// </para>
+    /// <para>
+    /// If database creation succeeds then the connection is changed to use the original database, if creation fails then
+    /// the original exception is thrown.
+    /// </para>
+    /// </remarks>
+    public static void OpenWithCreate(this SqlConnection connection)
```
Comment (davidfowl) [inline src/Components/Aspire.Microsoft.Data.SqlClient/AspireSqlServerSqlClientExtensions.cs:39]: I mean this isn't public API we would expose right? This should be an ADO.NET thing.

Comment (eerhardt) [reply src/Components/Aspire.Microsoft.Data.SqlClient/AspireSqlServerSqlClientExtensions.cs:39]: I said the same in https://github.com/dotnet/aspire/issues/4160#issuecomment-2108101239.

(not sure where the conversation should happen - the issue or the PR)

Comment (mitchdenny) [reply src/Components/Aspire.Microsoft.Data.SqlClient/AspireSqlServerSqlClientExtensions.cs:39]: Do it in the issue. This was just so we had something concrete to talk about.

And I agree this should be something in ADO.NET

**Comment by mitchdenny** [comment]
Closing this for now as this is probably not the route we will take. Further conversation will continue on the linked issue.

**Pull request closed without merging**